server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -19,11 +19,41 @@ def __init__(self, data, graph_method="umap", diffexp_method="ttest"):
         self._add_mandatory_annotations()
         self.cell_count = self.data.shape[0]
         self.gene_count = self.data.shape[1]
+        self._create_schema()
         self.graph_method = graph_method
         self.diffexp_method = diffexp_method
 
-    def _set_cell_names(self):
-        self.data.obs["cell_name"] = list(self.data.obs.index)
+    def _create_schema(self):
+        self.schema = {
+            "dataframe": {
+                "nObs": self.cell_count,
+                "nVar": self.gene_count,
+                "type": str(self.data.X.dtype)
+            },
+            "annotations": {
+                "obs": [],
+                "var": []
+            }
+        }
+        for ax in Axis:
+            curr_axis = getattr(self.data, str(ax))
+            for ann in curr_axis:
+                ann_schema = {"name": ann}
+                data_kind = curr_axis[ann].dtype.kind
```

Comment:
why use `dtype.kind`?    (see later comment about the ambiguity of `kind` for numbers).

I suggest doing something more explicit, such as:
```
data_type = curr_axis[ann].dtype
if data_type is np.float32:
    ...
elif data_type is np.int32:
    ...
```

Reply:
Summary of offline discussion:
- We want scientists to be able to load data into cellxgene with minimal preparation of their data so we aren't going to limit to a specific datatype for floats and ints
- For now, we will validate their data and warn them if they are using an unsupported datatype 
- Jeremy and Marcus will do some homework to see if large datatypes are actually being used in the wild and whether anyone is using them to their full precision or if they can be downcast without consequences